src/Controls/src/Core/Handlers/Items/iOS/TemplatedCell.cs

```diff
@@ -163,18 +173,10 @@ public void Bind(DataTemplate template, object bindingContext, ItemsView itemsVi
 				// Same template
 				if (oldElement != null)
 				{
-					if (oldElement.BindingContext == null || !(oldElement.BindingContext.Equals(bindingContext)))
```

Comment:
As stated in the comment above, it will always be clear. Either it's a new view, or one that has gone through the `PrepareForReuse` cleanup (which unregisters the callbacks).